File: vi_timing/LuaVsPython/sample.py
global_string = "global string"

def empty_func():
    pass

def bubble_sort(t, cmp = None):
    if not cmp:
#       cmp = lambda l, r: l < r
        cmp = c_ascending
    a = list(t)
    swapped = True
    while swapped:
        swapped = False
        for i in range(1, len(a)):
            if cmp(a[i], a[i - 1]):
                a[i], a[i - 1] = a[i - 1], a[i]
                swapped = True
    return a

File: vi_timing/LuaVsPython/sample.lua
global_string = "global string"

function empty_func()
end

function bubble_sort(aa, cmp)
    cmp = cmp or function(l, r) return l < r end

    local a = {table.unpack(aa)}
    local swapped = true
    while swapped do
        swapped = false
        for i = 2, #a do
            if cmp(a[i], a[i - 1]) then
                a[i], a[i - 1] = a[i - 1], a[i]
                swapped = true
            end
        end
    end
    return a
end
